File: ios-swiftui/library/Sources/PlanetComponents/TspEmpty.swift
import SwiftUI

public struct TspEmpty: View {
    let title: String
    let message: String
    let actionText: String
    let theme: StarPlanetTheme
    public init(title: String, message: String, actionText: String = "", theme: StarPlanetTheme = .sky) {
        self.title = title
        self.message = message
        self.actionText = actionText
        self.theme = theme
    }
    public var body: some View {
        VStack(spacing: 8) {
            Text("○").foregroundColor(.white).frame(width: 50, height: 50).background(theme.brandPrimary).clipShape(RoundedRectangle(cornerRadius: 18))
            Text(title).fontWeight(.bold).foregroundColor(theme.textPrimary)
            Text(message).foregroundColor(theme.textSecondary)
            if !actionText.isEmpty { TspButton(actionText, variant: .primary, theme: theme) }
        }
        .padding(24)
        .frame(maxWidth: .infinity)
        .background(theme.surfaceRaised)
        .overlay(RoundedRectangle(cornerRadius: 22).stroke(theme.borderDefault, lineWidth: 1))
        .clipShape(RoundedRectangle(cornerRadius: 22))
    }
}

File: flutter/library/lib/src/tsp_empty.dart
import 'package:flutter/material.dart';
import 'star_planet_theme.dart';
import 'tsp_button.dart';

class TspEmpty extends StatelessWidget {
  const TspEmpty({super.key, required this.title, required this.message, this.actionText = '', this.theme = StarPlanetTheme.sky, this.onAction});
  final String title;
  final String message;
  final String actionText;
  final StarPlanetTheme theme;
  final VoidCallback? onAction;

  @override
  Widget build(BuildContext context) => Container(
        width: double.infinity,
        padding: const EdgeInsets.all(24),
        decoration: BoxDecoration(
          color: theme.surfaceRaised,
          borderRadius: BorderRadius.circular(22),
          border: Border.all(color: theme.borderDefault),
        ),
        child: Column(children: [
          Container(
            width: 50,
            height: 50,
            alignment: Alignment.center,
            decoration: BoxDecoration(color: theme.brandPrimary, borderRadius: BorderRadius.circular(18)),
            child: const Text('○', style: TextStyle(color: Colors.white, fontSize: 22, fontWeight: FontWeight.w800)),
          ),
          const SizedBox(height: 8),
          Text(title, style: TextStyle(color: theme.textPrimary, fontWeight: FontWeight.w900)),
          const SizedBox(height: 6),
          Text(message, textAlign: TextAlign.center, style: TextStyle(color: theme.textSecondary)),
          if (actionText.isNotEmpty) ...[
            const SizedBox(height: 12),
            TspButton(text: actionText, fullWidth: false, variant: TspButtonVariant.primary, theme: theme, onTap: onAction),
          ],
        ]),
      );
}
